### ISSUE TITLE
Convergence problems and strong discrepancies in spectrum for abundance tomography application

### ISSUE THREAD

**Issue by talytha** [tardis-sn/tardis#662]
I wanted to verify the results I obtained in my bachelor thesis ("Abundance Tomography of SN2013dy") with the current master version of Tardis. I found strong discrepancies in the resulting spectra:

![abundance_tom_old_vs_new](https://cloud.githubusercontent.com/assets/22961286/19564380/aa3f261a-96e3-11e6-9f27-728f266c7ebd.png)

This seems to be caused by a much higher photospheric temperature in the current version compared to the old one. Moreover, the current version does not seem to converge during the 20 iterations which are set up. More details about this problem and the set up will follow.

**Comment by ftsamis** [comment]
Could you please include your configuration file?

**Comment by talytha** [comment]
We are pretty certain that the **old** Tardis version is f0bbeb72169a9171a928b0a774aad70d4d7e8e87.

The setup for the problem is in the attached zip file.

[config.zip](https://github.com/tardis-sn/tardis/files/542120/config.zip)

**Comment by talytha** [comment]
Looking at the config YML file, it looks like we are using ``hold_iterations`` (which used to be called ``hold``). So, this feature is broken, right @ftsamis?

**Comment by ftsamis** [comment]
The critical question: In your old run had you also renamed hold to hold_iterations and convergence to convergence_strategy in the configuration file?

If you didn't, you essentially ran tardis with the default convergence strategy mode which is `damped`.

The reason for that is that `convergence_criteria` was an invalid keyword at f0bbeb7 too, but what bit you was that the config_validator at that time didn't complain about extra keywords to inform you that convergence_criteria was not recognized, but instead it silently ignored them.

What I did was run your configuration with current **master** two times, one with the convergence_strategy you **thought** you were running tardis (blue) and one with completely omitting the convergence_strategy section (green), effectively setting it to the default damped mode. What I got is probably similar to your graph:
![figure_1](https://cloud.githubusercontent.com/assets/87870/19577578/8595d894-9720-11e6-8512-4363ab9d33f6.png)
 

And yes you are right, tardis_example was, until recently, using convergence_criteria too. (See #577)

For reference the default convergence_strategy is:

```
{'damping_constant': 0.5,
 'lock_t_inner_cycles': 1,
 't_inner': {'damping_constant': 0.5},
 't_inner_update_exponent': -0.5,
 't_rad': {'damping_constant': 0.5},
 'type': 'damped',
 'w': {'damping_constant': 0.5}}

```

**Comment by ftsamis** [comment]
With all that being said, hold_iterations is broken too, but in a more obvious way: it sometimes falls into an infinite loop, so it unfortunately doesn't have to do with it in your case.

**Comment by unoebauer** [comment]
Hey, @ftsamis - thanks for clearing this up! Looks like this solves the mystery! In the following, I am showing two plots, comparing two calculations with the current master (802995170c1bceda3861649ae1df3a18eb0978c0) with the old reference calculation. Once, I am using the specific convergence strategy, which was in the old config (but apparently was ignored by Tardis) and in the other comparison, I am showing a calculation with the default, damped, convergence strategy:

![tardis_cmp_1](https://cloud.githubusercontent.com/assets/6351165/19592437/1021e6d2-977c-11e6-8f5b-9b34915d2bac.png)

![tardis_cmp_2](https://cloud.githubusercontent.com/assets/6351165/19592444/13fbb0da-977c-11e6-97be-d1227f829915.png)

This looks very encouraging! In the second run, the BB temperature at the photosphere (15300K) is now also much closer to the one found in the original Tardis run (~14600K).

**Comment by ftsamis** [comment]
I have no idea why this looks encouraging, but I'm glad I helped. Does this mean we can close this (non?-)issue?

**Comment by unoebauer** [comment]
Let's see what other people think before closing.

(@ftsamis - if you look at the second plot you'll see that given that the same convergence strategy is used, the results of the new Tardis version match the old calculations quite well)

**Comment by unoebauer** [comment]
Some clarification is in order - we looked at the old TARDIS folders and modules again and it seems that we misidentified the TARDIS version we were using during the Bachelor thesis. It is **not**  f0bbeb7! 

Unfortunately, we only have the egg module lying around and there were problems with the ``githash`` attribute of the TARDIS module on the MPA system before. So I am not sure, if we can easily identify the version which was used...
